fsspec/mapping.py

```diff
@@ -34,9 +36,8 @@ class FSMap(MutableMapping):
 
     def __init__(self, root, fs, check=False, create=False, missing_exceptions=None):
         self.fs = fs
-        self.root = fs._strip_protocol(root).rstrip(
-            "/"
-        )  # we join on '/' in _key_to_str
+        self.root = fs._strip_protocol(root).rstrip("/")
+        self._root_key_to_str = fs._strip_protocol(os.path.join(root, "x"))[:-1]
```

Comment:
Will `os.path` not do the wrong thing on windows? How about directly using `posixpath.join`?
(sorry for all the comments on this change!)

Reply:
Good question. I made the change now.
tbh, i'm not sure though what happens with r"C:\abc\" as root or other windows style paths.
I'll try to spawn a windows VM and play around with it a bit.